processing/src/main/java/io/druid/query/groupby/GroupByQueryEngine.java

```diff
@@ -312,12 +314,113 @@ public RowIterator(GroupByQuery query, Cursor cursor, ByteBuffer metricsBuffer,
       dimensionSpecs = query.getDimensions();
       dimensions = Lists.newArrayListWithExpectedSize(dimensionSpecs.size());
       dimNames = Lists.newArrayListWithExpectedSize(dimensionSpecs.size());
+
       for (int i = 0; i < dimensionSpecs.size(); ++i) {
         final DimensionSpec dimSpec = dimensionSpecs.get(i);
-        final DimensionSelector selector = cursor.makeDimensionSelector(dimSpec.getDimension());
-        if (selector != null) {
-          dimensions.add(selector);
+        final String dimension = dimSpec.getDimension();
+        if(dimension.equals(Column.TIME_COLUMN_NAME)) {
```

Comment:
I agree that is logic seems out of place here. We should try to centralize the places that have to special case for the time column